app/models/secret_storage.rb

```diff
@@ -49,39 +50,65 @@ def self.keys
     end
   end
 
-  require 'vault'
   class HashicorpVault
 
     VAULT_SECRET_BACKEND ='secret/'.freeze
-    # we don't really want other directories in here,
+    # we don't really want other directories in the key,
     # and there may be other chars that we find we don't like
     ENCODINGS = {"/": "%2F"}
 
+    # get and cache a copy of the client that has a token
+    def self.vault_client
+      @vault_client ||= VaultClient.new
+    end
+
     def self.read(key)
-      result = Vault.logical.read(vault_path(key))
+      result = vault_client.logical.read(VAULT_SECRET_BACKEND + key)
       raise(ActiveRecord::RecordNotFound) if result.data[:vault].nil?
       result = result.to_h
       result = result.merge(result.delete(:data))
       result[:value] = result.delete(:vault)
       result
     end
 
+    # and parse it
     def self.write(key, data)
```

Comment:
this looks strange key parts are in the data ... can we generate the proper key before calling write ?